### ISSUE TITLE
Wildcard response status codes break TypeScript output

### ISSUE THREAD

**Issue by patrickarlt** [drwpow/openapi-typescript#1026]
**Description**

TypeScript is treating the numeric status codes as invalid TypeScript and is throwing error when wildcard status codes like `5XX` are used. You can see the output and errors [in the TypeScript playground](https://www.typescriptlang.org/play?#code/PQKhFgCgAIWgVAFgSwM7QGbIDYFNoDuAhukQK4AuA9gLQDmuAdrgE5EW4Am0ARgJ7QqAByZEhyGhT4jUAYxbIhFAHRRY0ACJVojKhWgBbIgGt8nZC1yz9sxEUYN01aBUT4seVTBDAofyLgAHkJULPrIjBwsGESy+ELsiOgA3mrQAETACa7pAFzQqTDQxQwU+cKs7MhUjKgA2ukcqBTpALoA3GkAvp2QXf5BIWEu0vgEuDyIVFTG6AC80ABKVqGcADzNCg4ANDq4AG6sAHy9UIOh4ZGsMXHQslQGIcyRKWlybkao+cyHLL3FllQT1QuC+BTSxVAcAAApxQfJFBRqoxoABBaCA4H4HwQ6AAZTIsjiqDBhWK5LuNQ4kXyZIp5PSYiE2GQsiqNWAACtUDU8tBNhE6P96T1caKipCwNBYfCFEpkZpcBQiDh0DiJdAAKIsFihWm44r3K408EaimMoTM1nsxhcnmMPkChzCini13Ct3QBJsAxK1hgn6sYWWACOZFBFAAQlRzKDvgcg2k3EQ4SwAwm-mlsogAJIcAzp369fqQM7BC4jETQIJRRhEbDQBbLe4sdZOui7QMsE4DcvDCJRG74CpsJE1V6lopNMqmimY8dx2f09IAJgADGu+fdHjUmBR6ul57VQW0GgSiaDUG0XQyAKwADXvW4eTz3B6PIKvrQa2t1LGvYoesWUBAA).

<img width="801" alt="2023-01-20_16-03-16" src="https://user-images.githubusercontent.com/378557/213827384-677f9d64-d454-493f-9a43-49e163408695.png">

| Name                 | Version                      |
|:---------------------|:-----------------------------|
| `openapi-typescript` | `6.10.0` - from `npx`        |
| Node.js              | `16.16.x`                      |
| OS + version         | `macOS 12` |

**Reproduction**

1. Convert the following spec.
2. Copy/paste the resulting TypeScript file into the TypeScript playground](https://www.typescriptlang.org/play)
3. See the error around the `5xx` status code

```yaml
openapi: 3.0.0
info:
  title: Test API
  version: 1.0.0
servers:
  - url: https://my-api.com
    description: Test
paths:
  /path:
    get:
      operationId: test
      responses:
        "200":
          $ref: "#/components/responses/Success"
        "5XX":
          $ref: "#/components/responses/Error"
components:
  responses:
    Success:
      description: A response
      content:
        application/json:
          schema:
            type: string
    Error:
      description: Details
      content:
        application/json:
          schema:
            type: string

```
**Expected result**

Response codes should probably be strings. The spec states that [the responses object](https://swagger.io/specification/#responses-object)s keys should always be quoted.

> This field MUST be enclosed in quotation marks (for example, "200") for compatibility between JSON and YAML. To define a range of response codes, this field MAY contain the uppercase wildcard character `X`. For example, `2XX` represents all response codes between `[200-299]`. 

It seems like the spec really intends for this to be a string. Searching around I think this might have been introduced in https://github.com/drwpow/openapi-typescript/pull/381?

**Checklist**

- [ ] I’m willing to open a PR (see [CONTRIBUTING.md](https://github.com/drwpow/openapi-typescript/blob/main/CONTRIBUTING.md))

**Comment by patrickarlt** [comment]
I downgraded to v5 to also work around https://github.com/drwpow/openapi-typescript/issues/984. v5 seems to properly format wildcard status codes as strings `"5XX"` and non-wildcard codes as numbers `200`.

**Comment by mitchell-merry** [comment]
Was this solved by #1010?

**Comment by mitchell-merry** [comment]
Looks like it:

```ts
export interface operations {

  test: {
    responses: {
      200: components["responses"]["Success"];
      "5XX": components["responses"]["Error"];
    };
  };
}

```
